template.yml

```diff
@@ -851,54 +852,57 @@ Resources:
               - logs:CreateLogGroup
               - logs:CreateLogStream
               - logs:PutLogEvents
-            Resource: 
-              !Join 
+            Resource: !Join
               - ""
               - - !Sub "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/"
-                - !Select [1, !Split [":function:", !GetAtt STMAnalyzeSegments.Arn]]
+                - !Select [
+                    1,
+                    !Split [
+                      ":function:",
+                      !GetAtt STMAnalyzeSegments.Arn,
+                    ],
+                  ]
                 - ":*"
 
-  STMAnalyzeSegmentsDataVersion:
-    Condition: SetSTMAnalyzeSegmentsData
+  STMAnalyzeSegmentsVersion:
+    Condition: STMAnalyzeSegmentsEnabled
     Type: AWS::Lambda::Version
-    DeletionPolicy: Retain
-    UpdateReplacePolicy: Retain
     Properties:
-      FunctionName: !Ref SetSTMAnalyzeSegmentsData
+      FunctionName: !Ref STMAnalyzeSegments
 
-  SetSTMAnalyzeSegmentsDataAlias:
-    Condition: SetSTMAnalyzeSegmentsData
+  STMAnalyzeSegmentsAlias:
+    Condition: STMAnalyzeSegmentsEnabled
     Type: AWS::Lambda::Alias
-    DeletionPolicy: Retain
-    UpdateReplacePolicy: Retain
     Properties:
-      FunctionName: !Ref SetSTMAnalyzeSegmentsData
-      FunctionVersion: !GetAtt SetSTMAnalyzeSegmentsDataVersion.Version
+      FunctionName: !Ref STMAnalyzeSegments
+      FunctionVersion: !GetAtt STMAnalyzeSegmentsVersion.Version
       Name: Live
   
-  SetSTMAnalyzeSegmentsDataTrigger:
+  STMAnalyzeSegmentsTrigger:
+    Condition: STMAnalyzeSegmentsEnabled
```

Comment:
The condition should not appear anywhere else than the alias and version blocks otherwise your function will be erased or launch errors when your condition is False.